src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsinterventionsservice/jpa/entity/Contract.kt

```diff
@@ -0,0 +1,63 @@
+package uk.gov.justice.digital.hmpps.hmppsinterventionsservice.jpa.entity
+
+import java.io.Serializable
+import java.time.OffsetDateTime
+import java.util.UUID
+import javax.persistence.Column
+import javax.persistence.Embeddable
+import javax.persistence.Embedded
+import javax.persistence.Entity
+import javax.persistence.EnumType
+import javax.persistence.Enumerated
+import javax.persistence.GeneratedValue
+import javax.persistence.Id
+import javax.persistence.JoinColumn
+import javax.persistence.ManyToOne
+import javax.persistence.PrimaryKeyJoinColumn
+import javax.persistence.SecondaryTable
+import javax.validation.constraints.NotNull
+
+enum class ContractType {
+  DYNAMIC_FRAMEWORK, STRUCTURED
+}
+
+enum class AgeGroup {
+  EIGHTEEN_TO_TWENTY_FIVE, TWENTY_FIVE_PLUS, BOTH
+}
+
+@Entity
+@SecondaryTable(name = "contract_detail", pkJoinColumns = [PrimaryKeyJoinColumn(name = "contract_id")])
+data class Contract(
+  @NotNull @Id @GeneratedValue val id: UUID? = null,
+  @Enumerated(EnumType.STRING)
+  @NotNull val type: ContractType,
+
+  @NotNull @ManyToOne @JoinColumn(name = "service_category_id")
+  val serviceCategory: ServiceCategory,
+
+  @NotNull @ManyToOne @JoinColumn(name = "service_provider_id")
+  val serviceProvider: ServiceProvider,
+
+  @NotNull val startDate: OffsetDateTime,
+  @NotNull val endDate: OffsetDateTime,
+
+  @Embedded
+  val contractDetail: ContractDetail,
+
+  @ManyToOne @JoinColumn(name = "region_id")
+  val region: Region? = null,
+
+  @ManyToOne @JoinColumn(name = "pcc_region_id")
+  val pccRegion: PCCRegion? = null,
+)
+
+@Embeddable
+data class ContractDetail(
```

Comment:
this is neat - but what's the reason these fields aren't just part of the contract entity ?

Reply:
I think the "optional" fields were pulled out separately from an initial suggestion.  Happy either way tbh